pkg/buses/reconciler.go

```diff
@@ -546,11 +547,24 @@ func (r *Reconciler) createOrUpdateBus(bus channelsv1alpha1.GenericBus) error {
 
 	// stash the new bus on the reconciler while retaining the old bus
 	bus, r.bus = r.bus, bus
+
 	if !equality.Semantic.DeepEqual(r.bus.GetSpec(), bus.GetSpec()) {
 		err := r.handler.onBus(r.bus, r)
 		if err != nil {
 			return err
 		}
+
+		// If parameters changed we need to reprovision / resubscribe
+		if !equality.Semantic.DeepEqual(r.bus.GetSpec().Parameters, bus.GetSpec().Parameters) {
+			r.logger.Infof("Bus parameters changed. Reprovisioning channels.")
+			for _, channel := range r.cache.AllChannels() {
```

Comment:
The Bus Parameters are subdivided into Channel and Subscription params. We could go a step further and check that the channel params have changed before requeueing all channels and that the subscription params have changed before requeueing all subscriptions.